google/cloud/security/inventory/pipelines/load_bigquery_datasets_pipeline.py

```diff
@@ -0,0 +1,175 @@
+            project_ids: A list of project ids.
+
+        Returns:
+            A list of objects like:
+                [[{'datasetId': 'test', 'projectId': 'bq-test'},
+                 {'datasetId': 'test', 'projectId': 'bq-test'}],
+                [{'datasetId': 'test', 'projectId': 'bq-test'},
+                 {'datasetId': 'test', 'projectId': 'bq-test'}]]
+
+        Raises: inventory_errors.LoadDataPipelineError when we encounter an
+        error in the underlying bigquery API.
+        """
+        dataset_project_map = []
+        for project_id in project_ids:
+            try:
+                result = self.api_client.get_datasets_for_projectid(project_id)
+            except api_errors.ApiExecutionError as e:
+                raise inventory_errors.LoadDataPipelineError(e)
+
+            if result:
+                dataset_project_map.append(result)
+
+        return dataset_project_map
+
+    def _retrieve_dataset_access_map(self, dataset_project_map):
+        """Iteriate through projects and their datasets to get ACLs.
+
+        Args:
+            dataset_project_map: A list of projects and their datasets.
+            See the output of _retrieve_dataset_project_map().
+
+        Returns:
+            A list of tuples in the form of:
+            [(project_id,
+              dataset_id,
+              {dataset_access_object}),
+            ...]
+        """
+        dataset_project_access_map = []
+        for map_item in dataset_project_map:
+            for item in map_item:
+                project_id = item.get('projectId')
+                dataset_id = item.get('datasetId')
+                dataset_acl = self._retrieve_dataset_access(project_id,
+                                                            dataset_id)
+
+                if dataset_acl:
+                    dataset_project_access_map.append(
+                        (project_id, dataset_id, dataset_acl)
+                        )
+
+        return dataset_project_access_map
+
+    def _transform(self, project_dataset_access_map):
+        """Yield an iterator of loadable groups.
+
+        Args:
+            A list of tuples in the form of:
+                [(project_id, dataset_id, {dataset_access_object}),...]
+
+        Yields:
+            An iterable of project_id, dataset_id, and access detail.
+        """
+
+        for (project_id, dataset_id, access) in project_dataset_access_map:
+            for acl in access:
+                yield {
+                    'project_id': project_id,
+                    'dataset_id': dataset_id,
+                    'access_domain': acl.get('domain'),
+                    'access_user_by_email': acl.get('userByEmail'),
+                    'access_special_group': acl.get('specialGroup'),
+                    'access_group_by_email': acl.get('groupByEmail'),
+                    'role': acl.get('role'),
+                    'access_view_project_id': acl.get(
+                        'view', {}).get('projectId'),
```

Comment:
sort nit: would it be easier to read if you did the following? (it's ok if you choose not to, just wondering about it)

```
view = acl.get('view', {})
...

view.get('projectId')
view.get('tableId')

etc.
```